Makefile

```diff
@@ -206,26 +201,67 @@ debug.conformance: manifests generate fmt vet
 
 ##@ Build
 
-.PHONY: build
-build: generate fmt vet ## Build manager binary.
+.PHONY: build.go
+build.go: generate fmt vet ## Build manager binary.
 	go build -o bin/manager main.go
 
-.PHONY: run
-run: manifests generate fmt vet ## Run a controller from your host.
+.PHONY: build.rust
+build.rust: ## Build dataplane
+	cargo xtask build-ebpf
+	cargo build
+
+.PHONY: build.rust.release
+build.rust.release: ## Build dataplane release
+	cargo xtask build-ebpf --release
+	cargo build --release
+
+.PHONY: run.go
+run.go: manifests generate fmt vet ## Run a controller from your host.
 	go run ./main.go
 
-.PHONY: debug
-debug: manifests generate fmt vet ## Run a controller from your host via debugger.
+.PHONY: debug.go
+debug.go: manifests generate fmt vet ## Run a controller from your host via debugger.
 	dlv debug ./main.go
 
-.PHONY: build.image
-build.image:
-	DOCKER_BUILDKIT=1 docker buildx build --platform=$(BUILD_PLATFORMS) $(BUILD_ARGS) -t $(BLIXT_CONTROLPLANE_IMAGE):$(TAG) .
+.PHONY: build.image.controlplane
+build.image.controlplane:
+	DOCKER_BUILDKIT=1 docker buildx build --platform=$(BUILD_PLATFORMS) --file=$(CONTROLPLANE_DOCKERFILE) $(BUILD_ARGS) -t $(BLIXT_CONTROLPLANE_IMAGE):$(TAG) .
+
+.PHONY: build.image.udp_server
+build.image.udp_server:
+	DOCKER_BUILDKIT=1 docker buildx build --platform=$(BUILD_PLATFORMS) --file=$(UDP_SERVER_DOCKERFILE) -t $(BLIXT_UDP_SERVER_IMAGE):$(TAG) .
+
+.PHONY: build.image.dataplane
+build.image.dataplane:
+	DOCKER_BUILDKIT=1 docker buildx build --platform $(BUILD_PLATFORMS) $(BUILD_ARGS) --file=$(DATAPLANE_DOCKERFILE) -t $(BLIXT_DATAPLANE_IMAGE):$(TAG) ./
 
 .PHONY: build.all.images
-build.all.images: build.image
-	cd dataplane/ && make build.image TAG=$(TAG)
-	cd tools/udp-test-server && make build.image TAG=$(TAG)
+build.all.images: 
+	$(MAKE) build.image.controlplane
+	$(MAKE) build.image.dataplane
+	$(MAKE) build.image.udp_server
+
+.PHONY: build.bytecode.images
+build.bytecode.images: build
+	docker build \
+	--build-arg PROGRAM_NAME=blixt-tc-ingress \
+	--build-arg BPF_FUNCTION_NAME=tc_ingress \
+	--build-arg PROGRAM_TYPE=tc \
+	--build-arg BYTECODE_FILENAME=loader \
+	-f https://raw.githubusercontent.com/bpfd-dev/bpfd/main/packaging/container-deployment/Containerfile.bytecode \
+	./target/bpfel-unknown-none/debug -t quay.io/bpfd-bytecode/blixt-tc-ingress:latest
+	docker build \
+	--build-arg PROGRAM_NAME=blixt-tc-egress \
+	--build-arg BPF_FUNCTION_NAME=tc_egress \
+	--build-arg PROGRAM_TYPE=tc \
+	--build-arg BYTECODE_FILENAME=loader \
+	-f https://raw.githubusercontent.com/bpfd-dev/bpfd/main/packaging/container-deployment/Containerfile.bytecode \
+	./target/bpfel-unknown-none/debug -t quay.io/bpfd-bytecode/blixt-tc-egress:latest
+
+.PHONY: push.bytecode.images
+push.bytecode.images: build.bytecode.images
+	docker push quay.io/bpfd-bytecode/blixt-tc-egress:latest
+	docker push quay.io/bpfd-bytecode/blixt-tc-ingress:latest
```

Comment:
not opposed to this, but I think that's outside the scope of this ticket. The original intention here was to merely migrate to cargo workspaces. Any changes you see in Dockerfile and otherwise were needed to accommodate this migration since it broke a lot of the CI/CD build & test pipeline.

Also @astoycos mentioned migrating completely to xtask maybe. Wouldn't that obviate further changes to the `Makefile`?

Reply:
This is fine I have plans to refactor this a bit 👍